pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java

```diff
@@ -97,9 +102,9 @@ public static String jsonFormat(Object object)
   @ScalarFunction
   public static Object jsonPath(Object object, String jsonPath) {
     if (object instanceof String) {
-      return JsonPath.read((String) object, jsonPath);
+      return PARSE_CONTEXT.parse((String) object).read(jsonPath, NO_PREDICATES);
```

Comment:
I am guessing this is threadsafe?

Reply:
yes, I don't know why it's allocated each time in the library, it just builds a `Configuration` each time (also thread-safe) which makes a wrapper around a Jackson `ObjectMapper` (thread-safe) available via `configuration.jsonProvider()`:

```java
    private final Configuration configuration;

    public ParseContextImpl() {
        this(Configuration.defaultConfiguration());
    }

    public ParseContextImpl(Configuration configuration) {
        this.configuration = configuration;
    }

    @Override
    public DocumentContext parse(String json) {
        notEmpty(json, "json string can not be null or empty");
        Object obj = configuration.jsonProvider().parse(json);
        return new JsonContext(obj, configuration);
    }
```

It might be best to fix this in the library but there's no guarantee on if or when that will get merged...

Reply:
Is there performance gain from passing in an empty array vs not passing the predicates?
```suggestion
      return PARSE_CONTEXT.parse((String) object).read(jsonPath);
```

Reply:
Yes, it prevents from allocating the array on each call. Alternatively we could try to add no varargs overloads to the library, which would be cleaner modulo the point above about guarantees.